src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,134 @@
 package core.basesyntax;
 
+import static java.util.Objects.hash;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public static final int DEFAULT_INITIAL_CAPACITY = 16;
+    public static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table = new Node[DEFAULT_INITIAL_CAPACITY];
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold) {
+            resize();
+        }
+        int index = getBucketIndex(key);
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, null);
+            size++;
+        }
+        if (table[index] != null
+                && (key != null && (!table[index].key.equals(key)))) {
+            Node<K, V> next = table[index].next;
+            if (next != null) {
+                while (next.next != null) {
+                    next = next.next;
+                }
+                next.next = new Node<>(key, value, null);
+                size++;
+            } else {
+                table[index].next = new Node<>(key, value, null);
+                size++;
+            }
+        }
+        if ((table[index]) != null && (table[index].key == key
+                || (key != null && key.equals(table[index].key)))) {
+            table[index].value = value;
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> e;
+        return (e = getNode(key)) == null ? null : e.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    final Node<K, V> getNode(Object key) {
+        int index = getBucketIndex(key);
+        int hash = (key == null) ? 0 : hash(key);
+        if (table != null && table.length > 0
+                && table[index] != null) {
+            if (getHashKey(table[index].key) == hash && (table[index].key == key
+                    || key != null && key.equals(table[index].key))) {
+                return table[index];
+            }
+            if ((table[index].next) != null) {
+                do {
+                    if (getHashKey(table[index].next.key) == hash
+                            && (table[index].next.key == key || (key != null
+                            && key.equals(table[index].next.key)))) {
+                        return table[index].next;
+                    }
+                    table[index].next = table[index].next.next;
+                } while ((table[index].next) != null);
+            }
+        }
+        return null;
```

Comment:
can be simplified. When we have bucket index, we need just one for/while loop, in which we will compare keys with equals
![image](https://github.com/mate-academy/jv-hash-map/assets/49767049/6ce85726-a404-4479-8d4a-6e33842e12be)